### PR TITLE
enable product & vendor production section fields

### PR DESCRIPTION
**Pull request by quol**

### CODE REVIEW

File: COT/edit_product.py [hunk 1]
```diff
@@ -112,6 +125,12 @@ def create_subparser(self, parent, storage):
         p.add_argument('-V', '--full-version',
                        help="""Software long version string, such as """
                        """"Cisco IOS-XE Software, Version 15.3(4)S" """)
+        p.add_argument('-p', '--product', metavar="PRODUCT",
```
Comment (glennmatthews) [inline COT/edit_product.py]: You shouldn't need to set `metavar` explicitly here or below, I think.

File: COT/ovf.py [hunk 2]
```diff
@@ -503,6 +503,43 @@ def system_types(self, type_list):
...
+        return None
+
+    @product.setter
+    def product(self, product_string):
+        if self.product_section is None:
+            self.product_section = self.set_or_make_child(
+                self.virtual_system, self.PRODUCT_SECTION)
+            # Any Section must have an Info as child
+            self.set_or_make_child(self.product_section, self.INFO,
+                                   "Product Information")
```
Comment (glennmatthews) [inline COT/ovf.py]: Now that we have four separate places where we might need to create PRODUCT_SECTION, I think it's time this got pulled out into a helper method - something like `def create_product_section_if_missing(self)`. Would you be willing to do this as part of your diffs?

**Comment by glennmatthews** [comment]
Please fix the following flake8 failures:

```
./COT/edit_product.py:82:80: E501 line too long (94 > 79 characters)
./COT/edit_product.py:114:80: E501 line too long (100 > 79 characters)
./COT/tests/test_edit_product.py:191:80: E501 line too long (81 > 79 characters)
./COT/ovf.py:523:5: E301 expected 1 blank line, found 0
./COT/ovf.py:585:5: E303 too many blank lines (2)
```

Other than this and the nitpicks above, this looks great - thanks for the contribution!

**Comment by quol** [comment]
NP thanks for the cool tool! I'll do the above tonight.

**Comment by quol** [comment]
I've done what was discussed, and added the other product attributes.  There are a couple of unit tests failing, test_wrapping & test_invalid_ovf.. I am not sure what the trouble is.

**Comment by quol** [comment]
some code regressed due to my pymode settings.  resolved.

**Comment by glennmatthews** [comment]
Thanks for the contribution! I merged your code into the develop branch and it will be included in the next release.

**Comment by quol** [comment]
I have something else for you take a look at when you have time.

https://github.com/quol/cot/commit/29b5e983ea71b23cd9b311665b9f951abe8508f9

This allowed me to avoid the costly copies that occur when using cot to
edit properties of an existing ovf.  The vmdks I'm working with are several
gb. Avoiding the copies of the vmdk files when simply editing properties of
the ovf descriptor shaves some precious minutes off my CI process.

I don't know what the utility of this is to you given your project goals,
so I didn't submit it.  I've thought about going a different approach, and
leveraging hard links if the copy to the working dir is not across devices,
etc.. but doing the above was the thinnest approach I could think of.

**Comment by glennmatthews** [comment]
Improving the performance of COT is definitely desirable and a number of the other workflows for COT are also in automated processes (notably, software build and packaging), so this is definitely of interest. I've offered a couple comments on the commit you pointed me to, but feel free to open an issue describing the specific concerns you have and we can take a look at them together.

Thanks!

**Comment by glennmatthews** [comment]
I have some ideas on how to optimize file write/copy operations, working on them as time permits. Do feel free to open an issue if you like.

**Pull request closed without merging**